packages/contracts/test/utils/assertions.ts

```diff
@@ -80,6 +98,10 @@ export async function expectInsufficientFundsAsync<T>(p: Promise<T>): Promise<vo
     return expect(p).to.be.rejectedWith(errMessage);
 }
 
+function escapeRegExp(str: string): string {
+    return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
```

Comment:
Not sure I fully understand what's going on here. Can you explain?

Reply:
If I want to match exactly the string s with a regex how do I do that?

`new RegExp(s)`? Not too fast cowboy. Some characters are special within the regexp context. Like back slashes, brackets, comas, questions marks etc. This function escapes all of that so if your string is `'.'` Your regex will match exactly that strings and not all strings with one character.
`'.'` -> `'\\.'`

Reply:
No worries, I didn't write that myself. it's from Mozilla docs. Scroll slightly up from here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Using_parentheses